### DISCUSSION TITLE
(COMPLETE) Feature Request: Sort or Search by Terrain

### DISCUSSION THREAD

**Discussion by Ugh-that-guy** [decheine/complete-compendium#208] [Ideas]
Hello!  I have almost no usable programming experience, but by god I'm going to come here and complain anyway! 🤣 
I'd love to help if possible, but I'm not sure that I meaningfully can (and hopefully the data are already present in the DB anyway).

The request is this:  I would love to be able to search the database for Monsters by RARITY (Common, Uncommon, Rare, Very Rare, Unique, Special, etc) or by TERRAIN (Forest, Mountain, et al).  This would be a huge help for me in prepping for my games!

**Comment by decheine** [comment]
Thank you for making this request, and I'm always happy to hear this site gets used for people's games.

That information IS available in the "database" (it's actually just one big JSON file so I don't have to pay to host anything). However, due to the nature of the 2e statblocks there can be many differently spelled or worded categories like for Climate/Terrain, which I anticipate will give like 50-100 different climate/terrains. This applies to the other stats too so I'll probably just start with climate. I'm considering using a field that you type into that shows a list of possible climates that you can click to add to the search query. 

Might incorporate design like this: https://m2.material.io/components/chips#anatomy

This will be what I work on when I get the time, likely this week.

**Reply by Ugh-that-guy** [reply]
Wow, thank you!

If nobody has told you this, this project is incredible. I run a kind of "crossroads of worlds" style game, so I get monsters from anywhere and everywhere.  This tool has been great to minimize lookups and cross references.  

Being able to quickly sort by rarity and terrain would let me develop encounter tables very quickly.

**Reply by decheine** [reply]
Thank you! I've seen around places on the internet like Mastodon people finding this site, I love bringing this to people. 

I looked into the Climates and Frequency's, and indeed there is an issue I expected. Among Climate's, even when I did some consolidating by splitting ones that have a '/' in them, there are over 1000 unique Climate/Terrain's. This problem isn't as bad for frequency, which has about 60 unique values. I have some UI ideas in mind though.

I should make an "advanced search" controls on the appendix, that also allows for a "details view" that lists monsters, and their monster subtype (like "Aartuk; Elder", "Aartuk; Warrior", etc) alongside their climate/terrain, frequency, and maybe some others like AC, Hit dice, XP. It's tricky because some of the monsters can be messy and have multiple entries for different hit dice of the same monster. 

I've had as a stretch goal a "Manual of the Planes" page, an overview of each of the planes and what creatures are native to them. This could include basically every plane, not just the outer planes, but every plane that is referenced in a "Climate/Terrain", so like Ravenloft, inner/outer planes, astral, etc. So I might prioritize that before completing the compendium, since there are a lot more books (mostly dragon magazines) that still need to be catalogued.

**Reply by decheine** [reply]
@Ugh-that-guy Sorry it's been several months now, but I just pushed the Table page. It has a fully customizable table view of the monsters, including split HD variants of hundreds of monsters, so the XP, THAC0, and HD of many monsters looks good. There are a few stragglers but only a few dozen. Filtering is limited to a simple text filter for now. Check it out when you get the chance.

**Reply by Ugh-that-guy** [reply]
I was only just today notified about this closure.  I checked the table, and it's exactly what I was hoping for as an easy solution.  Thank you!
